packages/e2e/test/web-extension/specs/wallet.spec.ts

```diff
@@ -45,15 +58,32 @@ describe('wallet', () => {
             return false;
           }
         });
-        await expect($(spanAddress)).toHaveTextContaining('addr');
+        walletAddr1 = await $(spanAddress).getText();
+        expect(walletAddr1).toHaveTextContaining('addr');
       });
       it('can build and sign a transaction', async () => {
-        await $(btnSignAndBuildTx).click();
-        await browser.waitUntil(async () => {
-          const signature = await $(divSignature).getText();
-          return signature.length > 1;
-        });
+        await buildAndSign();
+      });
+      it('can switch to another wallet', async () => {
+        await $(btnActivateWallet2).click();
+        await expect($(spanAddress)).not.toHaveTextContaining(walletAddr1);
+        await expect($(spanAddress)).toHaveTextContaining('addr');
       });
+
+      it('can build and sign a transaction using the new wallet', async () => {
+        await buildAndSign();
+      });
+
+      it('can switch back to the first wallet', async () => {
+        await $(btnActivateWallet1).click();
+        await expect($(spanAddress)).toHaveTextContaining(walletAddr1);
+      });
+
+      it('can deactivate the wallet and clear the stores', async () => {
+        await $(destroyWallets).click();
+        await expect($(spanAddress)).toHaveTextContaining('');
```

Comment:
This assertion probably matches everything
```suggestion
        await expect($(spanAddress)).toHaveText('');
```

Reply:
👍  Fixed in [7f4df72](https://github.com/input-output-hk/cardano-js-sdk/pull/502/commits/7f4df723ba6612846261661237effc7a1500ca80)